exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -236,25 +237,45 @@ func (prwe *prwExporter) execute(ctx context.Context, writeReq *prompb.WriteRequ
 	req.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
 	req.Header.Set("User-Agent", prwe.userAgentHeader)
 
-	resp, err := prwe.client.Do(req)
-	if err != nil {
-		return consumererror.NewPermanent(err)
-	}
-	defer resp.Body.Close()
+	retryInterval := time.Millisecond * 500 // Initial retry interval
+	maxRetryInterval := time.Second * 10     // Maximum retry interval
+	maxRetries := 5                          // Maximum number of retries
+	retryAttempts := 0
 
-	// 2xx status code is considered a success
-	// 5xx errors are recoverable and the exporter should retry
-	// Reference for different behavior according to status code:
-	// https://github.com/prometheus/prometheus/pull/2552/files#diff-ae8db9d16d8057358e49d694522e7186
-	if resp.StatusCode >= 200 && resp.StatusCode < 300 {
-		return nil
-	}
-	body, err := io.ReadAll(io.LimitReader(resp.Body, 256))
-	rerr := fmt.Errorf("remote write returned HTTP status %v; err = %w: %s", resp.Status, err, body)
-	if resp.StatusCode >= 500 && resp.StatusCode < 600 {
-		return rerr
+	for {
```

Comment:
for retrying you can use this package instead of implementing a retry strategy yourself: https://github.com/hashicorp/go-retryablehttp